google-cloud-firestore/src/test/java/com/google/cloud/firestore/QueryTest.java

```diff
@@ -303,6 +303,39 @@ public void inQueriesWithReferenceArray() throws Exception {
     assertEquals(expectedRequest, runQuery.getValue());
   }
 
+  @Test
+  public void inQueriesFieldsNotUsedInOrderBy() throws Exception {
```

Comment:
You may also want to validate that unary filters are handled properly and that inequalities result in an implicit order by if that's not already present.

Reply:
I think unary filters are basically isNull and isNan, which are equality checks by nature.